batch/batch_return.go

```diff
@@ -0,0 +1,48 @@
+package batch
+
+import "github.com/filecoin-project/go-state-types/exitcode"
+
+type BatchReturn struct {
+	SuccessCount uint64
+	FailCodes    []FailCode
+}
+
+type FailCode struct {
+	Idx  uint64
+	Code exitcode.ExitCode
+}
+
+func (b BatchReturn) Size() int {
+	return int(b.SuccessCount) + len(b.FailCodes)
+}
+
+func (b BatchReturn) AllOk() bool {
+	return len(b.FailCodes) == 0
+}
+
+func (b BatchReturn) Codes() []exitcode.ExitCode {
+	codes := make([]exitcode.ExitCode, b.Size())
+	i := 0
+	for _, fc := range b.FailCodes {
+		if fc.Idx > uint64(i) {
+			for ; i < int(fc.Idx); i++ {
```

Comment:
Here we use the invariant that FailCodes are in ascending order.  This is true for all BatchReturn constructed using BatchReturnGen which should be everything on chain.  I think this is a good assumption for this code.

I think we should consistently use this same invariant during traversal inside of CodeAt to stop traversal and return exitcode.OK if fc.Idx > n.

Reply:
Alternatively since batchreturn has a constant number of existcodes it might be overall better to use the less efficient but simple 

```golang

func (b BatchReturn) CodeAt(n uint64) exitcode.ExitCode {
     return b.Codes(n)
}
```

Reply:
@ZenGround0 do you have an opinion on maximum size of this thing? Can we max out at `math.MaxInt32` so we don't end up making supidly big slices? (even that's stupid-big but ..)